core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -969,13 +975,19 @@ private[spark] class Executor(
   /**
    * Download any missing dependencies if we receive a new set of files and JARs from the
    * SparkContext. Also adds any new JARs we fetched to the class loader.
+   * Visible for testing.
    */
-  private def updateDependencies(
+  private[spark] def updateDependencies(
```

Comment:
nit: Is `private[executor]` not sufficient ?

Reply:
It is. I updated this code to use `private[executor]` instead.